### PR TITLE
Optional Injection-less client side for SE

### PR DESCRIPTION
**Pull request by jansupol**
Signed-off-by: jansupol <jan.supol@oracle.com>

### CODE REVIEW

**Review by senivam — APPROVED** [review]

File: core-client/src/main/java/org/glassfish/jersey/client/innate/inject/NonInjectionManager.java [hunk 1]
```diff
@@ -0,0 +1,1030 @@
...
+
+        protected boolean _isPerThread(Class<? extends Annotation> scope) {
+            return RequestScoped.class.equals(scope) || PerThread.class.equals(scope);
+        }
+
+        protected X _getInstance(InstanceBinding<X> instanceBinding) {
+            return instanceBinding.getService();
+        }
+
+        protected X _create(SupplierInstanceBinding<X> binding) {
```
Comment (Verdent) [inline core-client/src/main/java/org/glassfish/jersey/client/innate/inject/NonInjectionManager.java:743]: should not these be private?

File: core-client/src/main/java/org/glassfish/jersey/client/innate/inject/NonInjectionManager.java [hunk 2]
```diff
@@ -0,0 +1,1030 @@
...
+            list.addAll(supplierClassBindings.stream()
+                    .map(this::_create)
+                    .collect(Collectors.toList()));
+
+            return list;
+        }
+
+        protected abstract X _create(SupplierClassBinding<X> binding);
+
+        protected abstract X _createAndStore(ClassBinding<X> binding);
```
Comment (Verdent) [inline core-client/src/main/java/org/glassfish/jersey/client/innate/inject/NonInjectionManager.java:808]: I thought _ was for "do not overwrite" methods

**Review by Verdent — CHANGES_REQUESTED** [review]

File: core-client/src/main/java/org/glassfish/jersey/client/innate/inject/NonInjectionManager.java (hunk 2 above)
Comment (jansupol) [reply core-client/src/main/java/org/glassfish/jersey/client/innate/inject/NonInjectionManager.java:808]: Methods with "_" are internal to its hierarchy, and not to be called from outside it.

File: core-client/src/main/java/org/glassfish/jersey/client/innate/inject/NonInjectionManager.java (hunk 1 above)
Comment (jansupol) [reply core-client/src/main/java/org/glassfish/jersey/client/innate/inject/NonInjectionManager.java:743]: Yep, nice catch

**Review by Verdent — APPROVED** [review]

**Comment by basil** [comment]
@jansupol This appears to have broken compatibility with RESTEasy 3.15.6 as demonstrated in the minimal reproducible example (MRE) at: https://github.com/MCMicS/jersey-jackson

Executed actions:

- Jersey 2.38: https://github.com/MCMicS/jersey-jackson/actions/runs/4337867360
- Jersey 2.39: https://github.com/MCMicS/jersey-jackson/actions/runs/4337871059

Due to this loss of compatibility the Jenkins project was forced to downgrade Jersey back to 2.38 in https://github.com/jenkinsci/jersey2-api-plugin/pull/60 and https://github.com/jenkins-infra/update-center2/pull/688.

**Comment by jansupol** [comment]
@basil Created a new [issue ](https://github.com/eclipse-ee4j/jersey/issues/5275)

**Pull request merged**